File: clients/java/examples/tables-create-text-index-object-analyzer.java
import static com.datastax.astra.client.core.lexical.AnalyzerTypes.STANDARD;

import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.core.lexical.Analyzer;
import com.datastax.astra.client.tables.Table;
import com.datastax.astra.client.tables.definition.indexes.TableTextIndexDefinition;
import com.datastax.astra.client.tables.definition.rows.Row;

public class Example {

  public static void main(String[] args) {
    // Get an existing table
    Table<Row> table =
        new DataAPIClient("**APPLICATION_TOKEN**")
            .getDatabase("**API_ENDPOINT**")
            .getTable("**TABLE_NAME**");

    // Index a column
    Analyzer analyzer =
        new Analyzer()
            .tokenizer(STANDARD.getValue())
            .addFilter("lowercase")
            .addFilter("stop")
            .addFilter("porterstem")
            .addFilter("asciifolding");
    TableTextIndexDefinition definition =
        new TableTextIndexDefinition().column("**TEXT_COLUMN_NAME**").analyzer(analyzer);
    table.createTextIndex("**INDEX_NAME**", definition);
  }
}

File: clients/go/examples/tables-create-text-index-object-analyzer.go
package main

import (
	"context"
	"log"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/options"
)

func main() {
	ctx := context.Background()

	// Get an existing table
	client := astra.NewClient()

	database := client.Database(
		"**API_ENDPOINT**",
		options.API().SetToken("**APPLICATION_TOKEN**"),
	)

	table := database.Table("**TABLE_NAME**")

	// Create a text index
	err := table.CreateTextIndex(
		ctx,
		"**INDEX_NAME**",
		"**TEXT_COLUMN_NAME**",
		options.CreateTextIndex().SetCustomAnalyzer(map[string]any{
			"tokenizer": map[string]any{
				"name": "standard",
				"args": map[string]any{},
			},
			"filters": []map[string]any{
				{"name": "lowercase"},
				{"name": "stop"},
				{"name": "porterstem"},
				{"name": "asciifolding"},
			},
			"charFilters": []map[string]any{},
		}),
	)
	if err != nil {
		log.Fatal(err)
	}
}
